app/assets/javascripts/short-form/ShortFormApplicationController.js.coffee

```diff
@@ -735,15 +729,22 @@ ShortFormApplicationController = (
 
   $scope.print = -> $window.print()
 
-  $scope.checkPrimaryApplicantAge = ->
+  $scope.checkAfterNamePage = ->
     if $scope.applicantDoesNotMeetSeniorRequirements()
       ShortFormNavigationService.isLoading(false)
       age = { minAge: $scope.listing.Reserved_community_minimum_age }
       $scope.eligibilityErrors = [$translate.instant('ERROR.SENIOR_EVERYONE', age)]
       $scope.handleErrorState()
     else
-      $scope.clearEligibilityErrors()
-      $scope.goToAndTrackFormSuccess('dahlia.short-form-application.contact')
+      if $scope.loggedIn()
+        $scope.goToAndTrackFormSuccess('dahlia.short-form-application.contact')
+      else
+        ShortFormNavigationService.isLoading(true)
+        AccountService.checkForAccount($scope.applicant.email).then ->
+          if AccountService.shortFormAccountExists()
```

Comment:
Wondering if we need a function to return this value? Or can we just say `AccountService.accountExists` since it gets assigned in `checkForAccount` anyway? Also just for naming, `shortFormAccountExists` makes it seems like there's more than one type of account.

Reply:
I believe used a function for a reason relating to making sure we got the current rather than initial value - @kaptron do you remember more about this?

Reply:
@callaghanc yeah that sounds familiar, I believe we were anticipating needing to use a $scope function in order to get the current value (i.e. rather than just saying $scope.value = Service.value which would just do a one-time assignment) however it looks like that wasn't really needed until chris's follow-up story here: https://github.com/Exygy/sf-dahlia-web/pull/955/files#diff-07eb5682f861e0fe07f9e3d209702a22R646

So yeah in the context of this `checkForAccount` function it could just be checking `AccountService.accountExists`.

Reply:
@kaptron @chrisdolendo If I changed this PR to use `AccountService.accountExists` and removed `AccountService.shortFormAccountExists()`, would that mess up your work that was based on this PR?

Reply:
@callaghanc should be fine -- the other branch could be merged/rebased to include any commits you make on this one.